### PR TITLE
Delete HttpClientDecorator

### LINKED ISSUE #912: Follow up tasks from PR #893
- [x] // Copy of HttpServerDecorator url handling (extract a method) in httpclienttracer
- [x] make spanNameForRequest protected in HttpClientTracer.java
- [x] update the package using tracer not decorator
- [x] clean up common code between client and server tracers
- [x] consider a new class for static methods extracted from BaseTracer
~~- [x] remove port and url in UrlInstrumentation. this requires updating tests. #1039.~~
- [x] rename and update unit tests under :instrumentation:agent-bootstrap:test
- [x] refactor startSpan(String spanName, Kind spanKind) in the BaseTracer.java
add startSpan(spanName) to clientTracer => Kind.CLIENT.
- [x] delete httpclientdecorator

### PR DESCRIPTION
**Pull request by heyams**
Fix #912

### CODE REVIEW

File: library-api/src/test/groovy/io/opentelemetry/instrumentation/api/tracer/HttpClientTracerTest.groovy [hunk 1]
```diff
@@ -195,6 +197,16 @@ class HttpClientDecoratorTest extends ClientDecoratorTest {
       protected String responseHeader(Map m, String name) {
         return m[name]
       }
+
+      @Override
+      protected HttpTextFormat.Setter<Map> getSetter() {
+        return null
```
Comment (anuraaga) [inline library-api/src/test/groovy/io/opentelemetry/instrumentation/api/tracer/HttpClientTracerTest.groovy:203]: Do we need to add any test to exercise this functionality?

**Comment by trask** [comment]
@heyams I merged upstream into this PR in order to resolve the merge conflicts I created via #970

File: library-api/src/test/groovy/io/opentelemetry/instrumentation/api/tracer/HttpClientTracerTest.groovy (hunk 1 above)
Comment (heyams) [reply library-api/src/test/groovy/io/opentelemetry/instrumentation/api/tracer/HttpClientTracerTest.groovy]: Per offline discussion with @trask, this is just cleaning up.  Adding tests will be a different task.

Comment (trask) [reply library-api/src/test/groovy/io/opentelemetry/instrumentation/api/tracer/HttpClientTracerTest.groovy]: @heyams is this tracked in #912?

**Comment by trask** [comment]
@heyams so sorry, I messed up and pushed your other branch to this PR when resolving conflicts, should be all good now

**Comment by heyams** [comment]
> @heyams so sorry, I messed up and pushed your other branch to this PR when resolving conflicts, should be all good now

No worries @trask.  Thanks for fixing it.

File: library-api/src/test/groovy/io/opentelemetry/instrumentation/api/tracer/HttpClientTracerTest.groovy (hunk 1 above)
Comment (heyams) [reply library-api/src/test/groovy/io/opentelemetry/instrumentation/api/tracer/HttpClientTracerTest.groovy]: Not yet.

Comment (trask) [reply library-api/src/test/groovy/io/opentelemetry/instrumentation/api/tracer/HttpClientTracerTest.groovy]: Ok, created #986

**Comment by iNikem** [comment]
> Fix #912

@heyams just a small note. Writing "fix #XXX" will close the issue on PR merge. As this PR fixes actually part of that issue, not the whole, it is better to write something like "Partially addresses #XXX" or "solves part of #XXX". This way you will have a reference between PR and issue, but will not trigger unwanted state changes.

**Pull request merged**